rust/chains/hyperlane-cosmos/src/aggregation_ism.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    grpc::{WasmGrpcProvider, WasmProvider},
+    payloads::aggregate_ism::{
+        ModulesAndThresholdRequest, ModulesAndThresholdRequestInner, ModulesAndThresholdResponse,
+    },
+    verify::bech32_decode,
+    ConnectionConf, CosmosProvider, Signer,
+};
+use async_trait::async_trait;
+use hyperlane_core::{
+    AggregationIsm, ChainResult, ContractLocator, HyperlaneChain, HyperlaneContract,
+    HyperlaneDomain, HyperlaneMessage, HyperlaneProvider, RawHyperlaneMessage, H256,
+};
+use tracing::instrument;
+
+/// A reference to an AggregationIsm contract on some Cosmos chain
+#[derive(Debug)]
+pub struct CosmosAggregationIsm {
+    _conf: ConnectionConf,
+    domain: HyperlaneDomain,
+    address: H256,
+    _signer: Signer,
+    provider: Box<WasmGrpcProvider>,
+}
+
+impl CosmosAggregationIsm {
+    /// create new Cosmos AggregationIsm agent
+    pub fn new(conf: ConnectionConf, locator: ContractLocator, signer: Signer) -> Self {
+        let provider = WasmGrpcProvider::new(conf.clone(), locator.clone(), signer.clone());
```

Comment:
@tkporter mentioned this and I double checked, I think the only rpc used from the config is the "grpcUrl", with an unecesarry "rpcUrl" still specified because that's not optional. Don't think it's a good idea to overload the "rpcUrl" with a grpc entry, but maybe the former could be made optional?